docs/user_guide/version-dropdown.rst

```diff
@@ -29,6 +29,12 @@ The switcher requires the following configuration steps:
 
 Below is a more in-depth description of each of these configuration steps.
 
+.. warning::
+    Modern web browsers do not allow loading data when the request is made from a page loaded via the ``file:`` protocol. This means that if you build your documentation locally and then directly open one of its pages in the browser, **the version switcher will not be allowed to load its JSON data source** and you'll end up with an empty switcher menu. Possible work-arounds are:
+
+    1. View the locally-built files through a local webserver (Python provides a builtin module for this: https://docs.python.org/3/library/http.server.html).
+    2. Disabling your browser's security settings (either by passing a command-line flag when launching the browser, or through a browser add-on).
+
```

Comment:
Should we print that, or a link to that in the console when we see the `file://` protocol ?

Reply:
I considered doing a console.log but then I thought: pointing folks to this docs page is way easier than explaining how to view browser console. So to me seemed not worth it but I don't mind adding it if you think is

Reply:
Yes, that's what I meant. 

I was also going to point out, if we detect this we could literally fill the version switcher itself with this text; but that may be overkill.